### PR TITLE
feat: Add phone validator

### PR DESCRIPTION
**Pull request by TSenter**

### CODE REVIEW

File: packages/ember/src/utils/phone.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/** A phone number with a 1-4 digit line number */
+export const lineNumberInput = /^(\d{1,4})$/;
+
+/** A phone number with a 1-4 digit exchange code */
```
Comment (KeithClinard) [inline packages/ember/src/utils/phone.ts:4]:
```suggestion
/** A phone number with a 1-3 digit exchange code */
```

File: packages/ember/src/validation/validators/phone.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    return result;
+  }
+
+  checkCountryCode(value: string, options: PhoneOptions): ValidateFnResponse {
+    const { countryCode, validCountryCodes, invalidCountryCodes } = options;
+    const cc = getCountryCode(value);
+
+    if (countryCode !== 'allow' || validCountryCodes || invalidCountryCodes) {
```
Comment (KeithClinard) [inline packages/ember/src/validation/validators/phone.ts:122]: Is this wrapping if statement necessary?
I think the logic works exactly the same if it is removed.

File: packages/ember/src/validation/validators/phone.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      if (cc && countryCode === 'disallow') {
+        return {
+          key: 'nrg.validation.phone.countryCode.notAllowed',
+          value,
+        };
+      } else if (!cc && countryCode === 'require') {
```
Comment (KeithClinard) [inline packages/ember/src/validation/validators/phone.ts:128]: I think this `else if` could just be a disconnected `if`

File: packages/ember/src/validation/validators/phone.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+    return true;
+  }
+
+  checkAreaCode(value: string, options: PhoneOptions): ValidateFnResponse {
+    const { areaCode, validAreaCodes, invalidAreaCodes } = options;
+
+    const ac = getAreaCode(value);
+
+    if (areaCode !== 'allow') {
```
Comment (KeithClinard) [inline packages/ember/src/validation/validators/phone.ts:160]: ditto.
This wrapping `if` can be removed.

File: apps/ember-test-app/tests/unit/validators/phone-test.ts [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+
+    this.model.field = '5557890';
+
+    let result = validator.result;
+
+    assert.deepEqual(result, {
+      isValid: true,
+      isWarning: false,
+      message: undefined,
+    });
```
Comment (KeithClinard) [inline apps/ember-test-app/tests/unit/validators/phone-test.ts:82]: Can we wrap this in a function to aid readability and general dryness?
`assertValid(validator)`, `assertInvalid(validator, message)`, `assertWarning(validator, message)`

I find myself breaking flow to read each of the options passed to `deepEqual` each time.

**Review by KeithClinard — CHANGES_REQUESTED** [review]

File: apps/ember-test-app/tests/unit/validators/phone-test.ts (hunk 5 above)
Comment (TSenter) [reply apps/ember-test-app/tests/unit/validators/phone-test.ts:82]: Can I do that in a separate PR? That's a change I'd like to make in all the validator tests

**Review by KeithClinard — APPROVED** [review]

**Pull request merged**